test/es-module/test-esm-loader-obsolete-hooks.mjs

```diff
@@ -0,0 +1,12 @@
+// Flags: --experimental-loader ./test/fixtures/es-module-loaders/hooks-obsolete.mjs
+
+import { mustCall } from '../common/index.mjs';
+import assert from 'assert';
+
+process.on('warning', mustCall((warning) => {
```

Comment:
My guess for this error is that the warning event is emitted before this code is reached. A possible solution would be too spawn a new Node.js process and check that its stderr contains the warning.

Reply:
Hmm, this is seeming like quite a lot of hassle to get around limitations of the testing suite. The deprecation warnings will not be around for long; would it be acceptable to forego a test for them?

Reply:
Another way of doing it would be to add the test in `test/message` instead. A test is better than no tests, if you feel that's too much work for you, I can push one to your branch after you've done the rebase-thingy.

Reply:
Oh sure! If it's just a matter of `git mv`, I can do that 😜

I just didn't want to go thru the trouble of trying spawning a process that _might_ work, when the current (with an `await import()` after `process.on` should've already worked).

Do I need to do anything special beyond just moving the file?

Reply:
Add a `.out` file with the expected output. You can use `test/message/unhandled_promise_trace_warnings.js` and `test/message/unhandled_promise_trace_warnings.out` as a reference.

Reply:
🤯

Reply:
@aduh95 thanks—that worked! No idea how I was expected to figure out the message tests on my own (I see no documentation related to them, no any explanation within that directory nor any of its tests).